### DISCUSSION TITLE
Ensure proper codec is running

### DISCUSSION THREAD

**Discussion by jacobweldon** [Arkq/bluez-alsa#617] [Q&A] [answered]
How can I confirm that bluealsa is running using the aptX codec?

When viewing the output of 'journalctl -b' I see the following:

raspberrypi bluealsa[828]: bluealsa: [828] D: ../../src/bluez.c:771: Registering battery provider: /org/bluez/hci0/battery
raspberrypi bluealsa[828]: bluealsa: [828] D: ../../src/bluez.c:791: BlueZ battery provider support not available
raspberrypi bluealsa[828]: bluealsa: [828] D: ../../src/bluez.c:585: Creating media endpoint object: /org/bluez/hci0/A2DP/aptX/sink/1
raspberrypi bluealsa[828]: bluealsa: [828] D: ../../src/bluez.c:496: Registering media endpoint: /org/bluez/hci0/A2DP/aptX/sink/1
raspberrypi bluealsa[828]: bluealsa: [828] D: ../../src/bluez.c:585: Creating media endpoint object: /org/bluez/hci0/A2DP/aptX/sink/2
raspberrypi systemd[1]: Started Hostname Service.
raspberrypi bluetoothd[825]: Failed to set privacy: Rejected (0x0b)
raspberrypi bluealsa[828]: bluealsa: [828] D: ../../src/bluez.c:496: Registering media endpoint: /org/bluez/hci0/A2DP/aptX/sink/2
raspberrypi bluealsa[828]: bluealsa: [828] D: ../../src/bluez.c:585: Creating media endpoint object: /org/bluez/hci0/A2DP/SBC/sink/1
raspberrypi bluealsa[828]: bluealsa: [828] D: ../../src/bluez.c:496: Registering media endpoint: /org/bluez/hci0/A2DP/SBC/sink/1
raspberrypi bluetoothd[825]: Endpoint registered: sender=:1.201 path=/org/bluez/hci0/A2DP/aptX/sink/1
raspberrypi bluetoothd[825]: Endpoint registered: sender=:1.201 path=/org/bluez/hci0/A2DP/aptX/sink/2
raspberrypi bluealsa[828]: bluealsa: [828] D: ../../src/bluez.c:585: Creating media endpoint object: /org/bluez/hci0/A2DP/SBC/sink/2
raspberrypi bluealsa[828]: bluealsa: [828] D: ../../src/bluez.c:496: Registering media endpoint: /org/bluez/hci0/A2DP/SBC/sink/2
raspberrypi bluetoothd[825]: Endpoint registered: sender=:1.201 path=/org/bluez/hci0/A2DP/SBC/sink/1
raspberrypi bluealsa[828]: bluealsa: [828] D: ../../src/bluez.c:1137: Signal: org.freedesktop.DBus.ObjectManager.InterfacesAdded()
raspberrypi bluetoothd[825]: Endpoint registered: sender=:1.201 path=/org/bluez/hci0/A2DP/SBC/sink/2

Can someone help me make sense of this? It looks to me like its using both SBC and aptX. Thanks!

**Comment by arkq** [comment, marked as answer]
The log which you've provided says that BlueALSA service is ready do start A2DP connection with either SBC or AptX codec. The relevant part is:
```
raspberrypi bluealsa[828]: bluealsa: Registering media endpoint: /org/bluez/hci0/A2DP/SBC/sink/2
                                                                                       ^- codec name
```

___
However, there is a better way of getting such information. Since version 3.1.0, bluez-alsa project provides a tool called `bluealsa-cli`. It needs to be enabled with `--enable-cli` during configuration. This tool will work regardless of whether bluez-alsa was configured with debug enabled or not, e.g.:

```
$ bluealsa-cli status
Service: org.bluealsa
Version: v4.0.0-97-g222c034-dirty
Adapters: hci0
Profiles:
  A2DP-source : SBC AAC aptX aptX-HD FastStream LDAC    <- list of available (enabled during configuration and
  A2DP-sink   : SBC AAC aptX aptX-HD FastStream LDAC       selected with -p) codecs for given Bluetooth profile
  HFP-AG      : CVSD mSBC
```

Later, with the same tool, you can check which codec was actually selected for given PCM, e.g.:
```
$ bluealsa-cli list-pcms
/org/bluealsa/hci0/dev_D0_16_B4_25_BB_AE/a2dpsnk/source
...

$ bluealsa-cli info /org/bluealsa/hci0/dev_D0_16_B4_25_BB_AE/a2dpsnk/source
Device: /org/bluez/hci0/dev_D0_16_B4_25_BB_AE
Sequence: 0
Transport: A2DP-sink
Mode: source
Running: false
Format: S32_LE
Channels: 2
Sampling: 96000 Hz
Available codecs: SBC AAC aptX aptX-HD LDAC
Selected codec: LDAC
Delay: 15.0 ms
SoftVolume: true
Volume: L: 127 R: 127
Muted: L: false R: false
```

**Reply by jacobweldon** [reply]
Thank you so much!
